core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,643 @@
+            if (titleBlockCell instanceof DAnnotation) {
+              annotationsList.add((DAnnotation) titleBlockCell);
+            }
+          }
+        }
+      }
+    }
+    ((DSemanticDiagram) elementView).getEAnnotations().removeAll(annotationsList);
+  }
+
+  public void clearLineEAnnotation(DAnnotation titleBlock, DDiagram diagram) {
+    List<DAnnotation> eAnnotationsList = diagram.getEAnnotations();
+    List<DAnnotation> annotationsListToBeRemoved = new ArrayList<>();
+    for (DAnnotation eAnnotation : eAnnotationsList) {
+      if (eAnnotation.getReferences().contains(titleBlock)) {
+        annotationsListToBeRemoved.add(eAnnotation);
+        for (EObject reference : eAnnotation.getReferences()) {
+          if (reference instanceof DAnnotation) {
+            annotationsListToBeRemoved.add((DAnnotation) reference);
+          }
+        }
+      }
+    }
+    eAnnotationsList.removeAll(annotationsListToBeRemoved);
+  }
+
+  public void clearColumnEAnnotation(DAnnotation titleBlock, DDiagram diagram) {
+    String columnNumber = titleBlock.getDetails().get(COLUMN);
+    List<DAnnotation> eAnnotationsList = diagram.getEAnnotations();
+    List<DAnnotation> annotationsListToBeRemoved = new ArrayList<>();
+    DAnnotation parentTitleBlock = getParentTitleBlock(titleBlock, diagram);
+    for (EObject reference : parentTitleBlock.getReferences()) {
+      if (reference instanceof DAnnotation) {
+        List<EObject> lineReferences = ((DAnnotation) reference).getReferences();
+        for (EObject lineRef : lineReferences) {
+          if (lineRef instanceof DAnnotation) {
+            if (((DAnnotation) lineRef).getDetails().get(COLUMN).equals(columnNumber)) {
+              annotationsListToBeRemoved.add((DAnnotation) lineRef);
+            }
+          }
+        }
+      }
+    }
+    eAnnotationsList.removeAll(annotationsListToBeRemoved);
+  }
+
+  public List<Object> getTitleBlockCellContent(EObject diagram, EObject cell, EObject containerView) {
+    List<Object> list = new ArrayList<Object>();
+    if ((diagram instanceof DRepresentation)) {
+      if (cell instanceof DAnnotation) {
+        DAnnotation annotation = (DAnnotation) ((DNodeContainer) (containerView.eContainer().eContainer())).getTarget();
+        FeatureInterpreter interpreter = new FeatureInterpreter();
+        try {
+          String feature = ((DAnnotation) cell).getDetails().get(CONTENT);
+          if (feature != null) {
+            EObject objToEvaluate = diagram;
+            List<EObject> modelElements = annotation.getReferences().parallelStream()
+                .filter(x -> !(x instanceof DAnnotation)).collect(Collectors.toList());
+            if (!modelElements.isEmpty()) {
+              objToEvaluate = modelElements.get(0);
+            }
+            Object obj = interpreter.evaluate(objToEvaluate, feature);
+            if (obj != null) {
+              if (obj instanceof String) {
+                boolean EANNOTATION_PRESENT = false;

```

Comment:
only constans in upper case. rename this variable, cammel case. (eAnnotationPresent)

Reply:
Renamed.